schema.sql

```diff
@@ -12,3 +12,28 @@ weight_kg		DECIMAL
 ALTER TABLE animals
 ADD COLUMN species VARCHAR(20);
 
+CREATE TABLE owners (
+id INT GENERATED ALWAYS AS IDENTITY,
+full_name VARCHAR(50) NOT NULL,
+age INT NOT NULL,
+PRIMARY KEY (id)
+);
+
+CREATE TABLE species (
+id INT GENERATED ALWAYS AS IDENTITY,
+name VARCHAR(20) NOT NULL,
+	PRIMARY KEY (id)
+);
+
+ALTER TABLE animals
+DROP COLUMN species;
+ALTER COLUMN id ADD GENERATED ALWAYS AS IDENTITY,
+ADD COLUMN species_id INT, 
+ADD COLUMN owner_id INT ,
+ADD CONSTRAINT fk_species
+FOREIGN KEY (species_id) REFERENCES species (id)
+ON DELETE CASCADE,
+ADD CONSTRAINT fk_owner
+FOREIGN KEY (owner_id) REFERENCES owners (id)
+ON DELETE CASCADE;
```

Comment:
Kindly note that you cannot drop a column and add multiple columns and constraints in the same ALTER TABLE statement. You need to execute the DROP COLUMN statement separately from the ADD COLUMN and ADD CONSTRAINT statements. The code below is a suggestion to fix the issue😉

```suggestion
//First, drop the column
ALTER TABLE animals
DROP COLUMN species;

//Then, add the columns and constraints
ALTER TABLE animals
ADD COLUMN species_id INT, 
ADD COLUMN owner_id INT,
ADD COLUMN id INT GENERATED ALWAYS AS IDENTITY,
ADD CONSTRAINT fk_species
FOREIGN KEY (species_id) REFERENCES species (id)
ON DELETE CASCADE,
ADD CONSTRAINT fk_owner
FOREIGN KEY (owner_id) REFERENCES owners (id)
ON DELETE CASCADE;

```

Note that I also moved the ADD COLUMN statement for the id column after the species_id and owner_id columns, since it's usually a good practice to define the non-generated columns before the generated ones. Kindly make the necessary change😉.